pkg/kubelet/cm/numamanager/numamanger.go

```diff
@@ -66,75 +68,151 @@ func (m *numaManager) GetAffinity(podUID string, containerName string) NumaMask
 }
 
 func (m *numaManager) calculateNUMAAffinity(pod v1.Pod, container v1.Container) NumaMask {
- 	podNumaMask := NumaMask {
- 		Mask:		nil,
- 		Affinity:	false,
- 	}
- 	var numaResult []int64
- 	numaResult = append(numaResult, 11)
- 	for _, hp := range m.hintProviders {
-        for resource, amount := range container.Resources.Requests {
-            glog.Infof("Container Resource Name in NUMA Manager: %v, Amount: %v", resource, amount.Value())
-            numaMask := hp.GetNUMAHints(string(resource), int(amount.Value()))
-            if numaMask.Affinity && numaMask.Mask != nil {
-                podNumaMask.Affinity = true
-                glog.Infof("[numamanager] numaMask in calculateNUMAAffinity: %v", numaMask.Mask)
-                orderedMask := getBitCount(numaMask.Mask)
-                if numaResult = getNumaAffinity(orderedMask, numaResult); numaResult == nil {
-                    glog.Infof("[numamanager] NO Numa Affinity. Result %v", numaResult)
-                    return NumaMask {
-                        Mask:		nil,
-                        Affinity:	true,
-                    }
-                }
-            } else if numaMask.Affinity && numaMask.Mask == nil {
-                glog.Infof("[numamanager] NO Numa Affinity.")
-                return NumaMask {
-                    Mask:		nil,
-                    Affinity:	true,
-                }
-            }
+        var row,col int = 1, 1
+        nmTest := make([][]int64, row)
+        for i:= range nmTest {
+                nmTest[i] = make([]int64,col)
         }
- 	}
- 	podNumaMask.Mask = append(podNumaMask.Mask, numaResult[0])
- 	return podNumaMask
- 	
+
+	podNumaMask := NumaMask {
+                Mask:           nmTest,
+                Affinity:       false,
+        }
+        
+	var maskHolder []string
+	count := 0 
+	var finalMask int64
+        for _, hp := range m.hintProviders {
+		for resource, amount := range container.Resources.Requests {
+			glog.Infof("Container Resource Name in NUMA Manager: %v, Amount: %v", resource, amount.Value())
+			numaMask := hp.GetNUMAHints(string(resource), int(amount.Value()))                          
+			if numaMask.Affinity && numaMask.Mask != nil {
+				if count == 0 {

```

Comment:
Maybe this could be a method too to calculate the MaskHolder?

Reply:
buildMaskHolder function added.